functions/Get-DbaDbTLogSpace.ps1

```diff
@@ -0,0 +1,118 @@
+
+    .PARAMETER SqlCredential
+        Login to the target instance using alternate Windows or SQL Login Authentication. Accepts credential objects (Get-Credential).
+
+    .PARAMETER Database
+        The database(s) to process - this list is auto-populated from the server. If unspecified, all databases will be processed.
+
+    .PARAMETER ExcludeDatabase
+        The database(s) to exclude - this list is auto-populated from the server.
+
+    .PARAMETER ExcludeSystemDatabase
+        Allows you to suppress output on system databases.
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Database, LogFile
+        Author: Jess Pomfret, JessPomfret.com
+
+        Website: https://dbatools.io
+        Copyright: (c) 2019 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Get-DbaDbTLogSpace
+
+    .EXAMPLE
+        PS C:\> Get-DbaDbTLogSpace -SqlInstance Server1
+
+        Returns the transaction log usage information for all databases on Server1
+
+    .EXAMPLE
+        PS C:\> Get-DbaDbTLogSpace -SqlInstance Server1 -Database Database1, Database2
+
+        Returns the transaction log usage information for both Database1 and Database 2 on Server1
+
+    .EXAMPLE
+        PS C:\> Get-DbaDbTLogSpace -SqlInstance Server1 -ExcludeDatabase Database3
+
+        Returns the transaction log usage information for all databases on Server1, except Database3
+
+    .EXAMPLE
+        PS C:\> Get-DbaDbTLogSpace -SqlInstance Server1 -ExcludeSystemDatabase
+
+        Returns the transaction log usage information for all databases on Server1, except the system databases
+
+    .EXAMPLE
+        PS C:\> Get-DbaRegisteredServer -SqlInstance cmsServer | Get-DbaDbTLogSpace -Database Database1
+
+        Returns the transaction log usage information for Database1 for a group of servers from SQL Server Central Management Server (CMS).
+
+    #>
+    [CmdletBinding()]
+    param (
+        [parameter(ValueFromPipeline, Mandatory)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [string[]]$Database,
+        [string[]]$ExcludeDatabase,
+        [switch]$ExcludeSystemDatabase,
+        [switch]$EnableException
+    )
+
+    process {
+        foreach ($instance in $SqlInstance) {
+            try {
+                $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential
```

Comment:
Do we need to validate versions that support the output or is it all the same from 2005+?

I wonder if we need to spit warning if the server passed in is for Azure SQL? The command will run when you are on Azure SQL but it won't return information about user databases. It only works against the system databases.

Reply:
It is the same from 2008+, I don't have a 2005 box to test with. @potatoqualitee do you have 2005 in your lab?